src/storage/src/hummock/store/version.rs

```diff
@@ -609,21 +611,61 @@ impl HummockVersionReader {
-                    if sstable_info
-                        .table_ids
-                        .binary_search(&read_options.table_id.table_id)
-                        .is_err()
-                    {
-                        continue;
-                    }
-                    let sstable = self
-                        .sstable_store
-                        .sstable(sstable_info, &mut local_stats)
+                let fetch_meta_req = level.table_infos[start_table_idx..=end_table_idx]
+                    .iter()
+                    .filter(|sstable_info| {
+                        sstable_info
+                            .table_ids
+                            .binary_search(&read_options.table_id.table_id)
+                            .is_ok()
+                    })
+                    .collect_vec();
+                fetch_meta_reqs.push((level.level_type, fetch_meta_req));
+            } else {
+                let table_infos = prune_ssts(
+                    level.table_infos.iter(),
+                    read_options.table_id,
+                    &table_key_range,
+                );
+                if table_infos.is_empty() {
+                    continue;
+                }
+                // Overlapping
+                let fetch_meta_req = table_infos.into_iter().rev().collect_vec();
+                fetch_meta_reqs.push((level.level_type, fetch_meta_req));
+            }
+        }
+        let mut flatten_reqs = vec![];
+        let mut req_count = 0;
+        for (_, fetch_meta_req) in &fetch_meta_reqs {
+            for sstable_info in fetch_meta_req {
+                let inner_req_count = req_count;
+                let capture_ref = async {
+                    self.sstable_store
+                        .sstable_syncable(sstable_info, &local_stats)
                         .in_span(Span::enter_with_local_parent("get_sstable"))
-                        .await?;
+                        .await
+                };
+                flatten_reqs
+                    .push(async move { capture_ref.await.map(|result| (inner_req_count, result)) });
+                req_count += 1;
+            }
+        }
+        let mut flatten_resps = vec![None; req_count];
+        let mut buffered = stream::iter(flatten_reqs).buffer_unordered(10);
+        while let Some(result) = buffered.next().await {
+            let (req_index, resp) = result?;
+            flatten_resps[req_count - req_index - 1] = Some(resp);
+        }
+        drop(buffered);
 
+        for (level_type, fetch_meta_req) in fetch_meta_reqs {
+            if level_type == LevelType::Nonoverlapping as i32 {
+                let mut sstables = vec![];
+                for sstable_info in fetch_meta_req {
+                    let (sstable, local_cache_meta_block_miss) =
+                        flatten_resps.pop().unwrap().unwrap();
+                    local_stats.apply_meta_fetch(local_cache_meta_block_miss);
```

Comment:
Can we add a assertion to make sure the sstable matches?
```
assert_eq!(sstable_info.id, sstable.value().id);
```

Reply:
> Can we add a assertion to make sure the sstable matches?
> 
> ```
> assert_eq!(sstable_info.id, sstable.value().id);
> ```

fixed